scripts/scripts.js

```diff
@@ -17,6 +17,20 @@
  */
 
 export function sampleRUM(checkpoint, data = {}) {
+  sampleRUM.defer = sampleRUM.defer || [];
+  const defer = (fnname) =>
+    sampleRUM[fnname] = sampleRUM[fnname]
+    || ((...args) => sampleRUM.defer.push({ fnname, args }));
+  sampleRUM.drain = sampleRUM.drain ||
+    ((fnname, fn) => {
+      sampleRUM[fnname] = fn;
+      sampleRUM.defer
+        .filter(({ fnname: dfnname }) => dfnname === fnname)
+        .forEach(({ fnname, args }) => sampleRUM[fnname](...args));
+    });
+  sampleRUM.on = (chkpnt, fn) => sampleRUM.cases[chkpnt] = fn;
```

Comment:
This is a net new extension point that allows overriding the pinging behaviour for a given checkpoint. 

Usage:

```javascript
sampleRUM.on('mycheckpoint', (ping, data) => {
  console.debug('mycheckpoint triggered');
  ping({ ...data, pong: true });
}
```